src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -81,41 +81,54 @@ public static Task CopyToEndAsync(this IPipeReader input, Stream stream)
         {
             try
             {
-                await input.CopyToAsync(stream, bufferSize, cancellationToken);
+                // TODO: Use bufferSize argument
+                while (!cancellationToken.IsCancellationRequested)
```

Comment:
If this method is going to take a CancellationToken, it makes sense to register a cancellation callback that calls CancelPendingRead() in case the IPipeReader is idle.